pkg/provisioner/ironic/testserver/ironic.go

```diff
@@ -76,23 +76,46 @@ func (m *IronicMock) WithDrivers() *IronicMock {
 	return m
 }
 
-// WithNode configures the server with a valid response for /v1/nodes
-func (m *IronicMock) WithNode(node nodes.Node) *IronicMock {
+func (m *IronicMock) buildURL(url string, method string) string {
+	return fmt.Sprintf("%s:%s", url, method)
+}
+
+func (m *IronicMock) withNode(node nodes.Node, method string) *IronicMock {
```

Comment:
As this is a private method, maybe we can rename it into something longer but easier to differentiate from `WithNode`?

Reply:
Unless it's a convention I'm not aware of.

Reply:
No particular convention. Do you have any specific idea for the name?

Reply:
I think that I would leave the generic method as it is but rename the specific/partial ones to `WithGetNode`, `WithPatchNode` and so on. Though I'm not sure how compatible it is.

I must agree that choosing a name here is a bit difficult, since the functions do HTTP requests internally, but there's an abstraction level for the user.

Or maybe something like `genericWithNode`? What do you think?

Reply:
maybe something like `withNodeBase`, being a base technical method (but honestly don't have better ideas)